Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -140,6 +143,14 @@
   public long maxMillisToWaitForServerResponse = 0L;
   @NotNull private final InteractiveErrorReporter myErrorReporter = new InteractiveErrorReporter();
 
+  @NotNull private final Object myDebugLogLock = new Object();
+  @NotNull private final LinkedHashMap<String, String> myDebugLog = new LinkedHashMap<String, String>(DEBUG_LOG_CAPACITY) {
```

Comment:
It was StackOverflow where I found EvictingQueue :). Queue/List works O(1) and its algorithm is very simple whereas LinkedHashMap adds some overhead. Not a big deal for our case, just `put(str, str)` looked a bit unnatural to me.